sdks/python/apache_beam/io/requestresponse.py

```diff
@@ -411,3 +384,430 @@ def process(self, request: RequestT, *args, **kwargs):
   def teardown(self):
     self._metrics_collector.teardown_counter.inc(1)
     self._caller.__exit__(*sys.exc_info())
+
+
+class Cache(abc.ABC):
+  """Base Cache class for
+  :class:`apache_beam.io.requestresponse.RequestResponseIO`.
+
+  For adding cache support to RequestResponseIO, implement this class.
+  """
+  @abc.abstractmethod
+  def get_read(self):
+    """get_read returns a PTransform that reads from the cache."""
+    pass
+
+  @abc.abstractmethod
+  def get_write(self):
+    """get_write returns a PTransform that writes to the cache."""
+    pass
+
+  @abc.abstractmethod
+  def has_request_coder(self) -> bool:
+    """returns `True` if the request coder is present."""
+    pass
+
+  @abc.abstractmethod
+  def set_request_coder(self, request_coder: coders.Coder):
```

Comment:
I like these python's getter-setter but in this case since i'm just pushing it to the cache class I would like to avoid it.  https://stackoverflow.com/questions/17576009/python-class-property-use-setter-but-evade-getter

Reply:
I think logic would be easier to follow and less verbose if you make the coders to be regular (public) attributes, set them in the constructor and keep immutable or maybe default to StrUtf8Coder if not specified by the user. I don't want to start bikeshedding on this though.

Reply:
I agree, made these as property